README.md

```diff
@@ -105,6 +105,12 @@ $ minikube addons disable metrics-server
 
 The following versions are supported and work as we test against these versions in their respective branches. But note that other versions might work!
 
+<!---
+**NOTE WHEN UPDATING THE MATRIX**: make sure to also apply the changes to the CI workflows.
+1. Update the [e2e-tests job](/.github/workflows/ci.yaml) to test HEAD against all compatible Kubernetes versions.
+2. Update the list of supported release branch in the [upgrade workflow](/.github/workflows/versions.yaml).
```

Comment:
What do you think about creating `RELEASE.md` doc and put this info there?

Reply:
sounds good